### ISSUE TITLE
[MENFORCER-211] wildcard ignore in requireReleaseDeps

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-enforcer#422]
**[Arne Brix](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=xophos)** opened **[MENFORCER-211](https://issues.apache.org/jira/browse/MENFORCER-211?redirect=false)** and commented

the bannedDependencies rule allows one to specify artifact lists using wildcards:

```xml
org.apache.*:maven-*:*
```

we are using the requireReleaseDeps rule for our release builds.
We would like to exclude dependencies with scope test because they don't contribute to the final artifact.
We tried the syntax documented for bannedDependencies, but it didn't work:

```xml
<configuration>
  <rules>
    <requireReleaseDeps>
      <message>No Snapshots Allowed!</message>
      <onlyWhenRelease>true</onlyWhenRelease>
      <excludes>
        <exclude>*:*:*:*:test</exclude>
      </excludes>
    </requireReleaseDeps>
  </rules>
  <fail>true</fail>
</configuration>
```

---

**Affects:** 1.3.1

0 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Can you provide a test project which shows the problem? Thanks in advance.

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Can you describe more in detail what you mean by ", but it didn't work" ?

**Comment by jira-importer** [comment]
**[Dariusz Kordonski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kordzik)** commented

I can confirm that wildcards mixed with strings don't seem to work. Pure wildcards as reported in this issue work for us however. E.g. both excludes below will work (exact match, pure wildcard):

```
<excludes>
   <exclude>com.atlassian.plugins:atlassian-plugins-core</exclude>
 </excludes>
```

```
<excludes>
   <exclude>com.atlassian.plugins:*</exclude>
 </excludes>
```

But the following will not be detected (mixed string and wildcards), despite being documented as a valid exclusion pattern:

```
<excludes>
   <exclude>com.atlassian.plugins:atlassian-plugins-*</exclude>
 </excludes>
```

We are using version 1.4.0

**Comment by jira-importer** [comment]
**[Dariusz Kordonski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kordzik)** commented

Actually disregard my comment, our case is `bannedDependencies` rule, which is different. I'll raise a separate bug.

**Comment by jira-importer** [comment]
**[Elliotte Rusty Harold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=elharo)** commented

seems useful

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [5409be83dc3b621121e6222ad3830f8e95cf6614](https://gitbox.apache.org/repos/asf?p=maven-enforcer.git;a=commit;h=5409be83dc3b621121e6222ad3830f8e95cf6614)
Documentation has been updated to explain what the pattern looks like.
